### PR TITLE
[16.0][ADD] product_optional_product_quantity: add module

### PR DESCRIPTION
**Pull request by dmitriypaulov**

### CODE REVIEW

File: product_optional_product_quantity/models/product_optional_line.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+        domain="[('company_id', 'in', [company_id, False])]",
+        string="Product",
+        required=True,
+    )
+    quantity = fields.Float(
+        default=1.0,
+        required=True,
+    )
+
+    @api.constrains("product_tmpl_id", "optional_product_tmpl_id")
```
Comment (angelinaanaki) [inline product_optional_product_quantity/models/product_optional_line.py:27]: I think you can also add a domain in both fields, for example for **optional_product_tmpl_id**:

`domain="[('id', '!=', product_tmpl_id)]"`

This way, the user will not be able to select the same product.

File: product_optional_product_quantity/models/product_template.py [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+        optional_quantity_enabled = self.env.user.has_group(
+            "product_optional_product_quantity.group_product_optional_quantity"
+        )
+        for r in self:
+            if optional_quantity_enabled:
+                r.optional_product_ids = r.product_optional_line_ids.mapped(
+                    "optional_product_tmpl_id"
+                )
+            else:
+                r.optional_product_ids = r.optional_product_ids
```
Comment (angelinaanaki) [inline product_optional_product_quantity/models/product_template.py]: The group is specifically assigned to the quantity field. However, in this scenario, why aren't the **optional_product_ids** being populated?

**Review by angelinaanaki — APPROVED** [review]
LGTM

**Review by DemchukM — APPROVED** [review]
LGTM

**Comment by ivs-cetmix** [comment]
hey @OCA/product-maintainers would appreciate someone having a look at this module ) It's a blocking dep for [other PR](https://github.com/OCA/sale-workflow/pull/2911)

**Pull request merged**